quality-assurance/test-code-standards.md

```diff
@@ -1,42 +1,54 @@
 # Test Automation Standards
 
-## C# and Java
+## Test Code
 
 * Test Automation code should adhere to the same standard as production code
 * Formatting standards should reflect standard in use by development teams (e.g. Google Code Standard, Microsoft)
 * Standard coding principles apply (e.g. SOLID, DRY)
 
 ## Unit Testing
-
-* Aim for naming consistency. Common standard is to use a When-Then name, for example, WhenTwoItemsExistsThenBothItemsAreReturned
-* Follow best practice:
-    * Tests for results not functionality
-    * One assertion per test
-    * Tests should be isolated, i.e. have no dependencies on other tests nor on order of execution
-
-## API Testing
-
-* Perform full happy/unhappy path tests at this level
-* Interactions with APIs should be abstracted into a separate service/facade, not alongside the test code
+### Test a single Method in isolation
+* Test naming should follow a convention agreed by the team. An example is **Method_Action_Assertion** e.g. ValidateObject_WhenObjectPropteryIsNull_ThrowsArgurmentException
+* Tests should be isolated, i.e. have no dependencies on other tests nor on order of execution.
+* Fast. Unit tests should take very little time to run. Milliseconds not seconds!
+* Isolated. Unit tests are standalone, can be run in isolation, and have no dependencies on any outside factors such as a file system or database.
+* Repeatable. Running a unit test should be consistent with its results,it always returns the same result if you do not change anything in between runs.
+* Self-Checking. The test should be able to automatically detect if it passed or failed without any human interaction.
+* Timely. A unit test should not take a disproportionately long time to write compared to the code being tested. If you find testing the code taking a large amount of time compared to writing the code, consider a design that is more testable.
+* Unit Test coverage should aim to cover around 80% of code.
+* Aim for one assertion per test.
+
+## Component Testing
+### Multiple methods in a single test
+* Component tests should follow the same standard as unit tests
+* The majority of test coverage should come from Unit Tests not Component Tests.
+* Try to test no more than two components per test. Prefer multiple tests over one long component test.
+* Aim for one assertion per test.
+
+## Integration Testing 
+### Multiple components integration with real dependencies (i.e. database integration)
+* Determine test data ahead of time and ensure this test data is reusable.
+* Ensure integration tests have as few external dependencies as possible outside of the test code.
+* Fakes, mocks and stubs are permitted, provided the test is asserting integration with a real dependency (e.g. database or message broker integration)
+* Reasonably fast. Seconds not minutes!
+* Aim to test one behavior per test.
+* Perform as many scenarios as possible at this level. Ensure scenarios not covered here are covered by unit/component tests.
```

Comment:
are we not going to point them into splitting the integration tests / API up into two categories. 

1. connectivity (Swagger or Open API) should be fast to build and execute
2. contents based tests where there is a need to confirm that the response is as expected.

there is then the ? of - do we risk base the test coverage as we are not aiming for 100% but enough to remove or reduce the risk to an acceptable level

Reply:
Added the bit around api tests in the same section. Let me know if it makes sense.

_there is then the ? of - do we risk base the test coverage as we are not aiming for 100% but enough to remove or reduce the risk to an acceptable level_
I think this is covered by line 35? let me know if not